### PR TITLE
Changes in documentation version's drop-down

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

This PR provides changes which are described within https://github.com/handsontable/handsontable/issues/10014. 

1. Patches were added to the drop-down with styles which have been approved by @krzysztofspilka.

<table>
<tr>
	<td>Light Theme
	<td>Dark Theme
<tr>
	<td>Screenshot 2022-10-28 at 17 24 18
	<td>Screenshot 2022-10-28 at 17 24 25
</table>

3. There were introduced an extra *patches* property for `common.json` file stored on the server.

~**Note:** The file stored on handsontable.com should be changed. There are mocks in the code.~

4. Number of minor versions has been reduced to `6`. The rest of the versions are still to be hosted.
5. The `Current` label was changed to a `Latest` text. Also a patch version is shown there.

*[skip changelog]*

### Related issue(s):
1. #10014

### CODE REVIEW

File: docs/.vuepress/plugins/dump-docs-data/docs-versions.js [hunk 1]
```diff
@@ -46,15 +51,29 @@ async function readFromGitHub() {
...
+      const minorVersion = `${semver.parse(tag).major}.${semver.parse(tag).minor}`;
+      const patchVersion = `${minorVersion}.${semver.parse(tag).patch}`;
 
-  const tags = Array.from(tagsSet);
+      tagsSet.add(minorVersion);
+
+      if (minorsToPatches.has(minorVersion)) {
+        const uniquePathes = Array.from(new Set(minorsToPatches.get(minorVersion)).add(patchVersion));
+
+        minorsToPatches.set(minorVersion, uniquePathes);
```
Comment (budnix) [inline docs/.vuepress/plugins/dump-docs-data/docs-versions.js:68]:
```suggestion
        const uniquePaths = Array.from(new Set(minorsToPatches.get(minorVersion)).add(patchVersion));

        minorsToPatches.set(minorVersion, uniquePaths);
```

File: docs/.vuepress/plugins/dump-docs-data/docs-versions.js [hunk 2]
```diff
@@ -66,6 +85,7 @@ async function readFromGitHub() {
   return {
     versions,
     latestVersion: versions[0],
+    patches,
```
Comment (budnix) [inline docs/.vuepress/plugins/dump-docs-data/docs-versions.js:88]: After removing the mocks, the generated patches are way beyond the limit of 6 MAJOR versions. Can you apply the limit for patches as well?
```json
{"patches":[["12.2",["12.2.0"]],["12.1",["12.1.3","12.1.2","12.1.1","12.1.0"]],["12.0",["12.0.1","12.0.0"]],["11.1",["11.1.0"]],["11.0",["11.0.1","11.0.0"]],["10.0",["10.0.0"]],["9.0",["9.0.2","9.0.1","9.0.0"]],["8.4",["8.4.0"]],["8.3",["8.3.2","8.3.1","8.3.0"]],["8.2",["8.2.0"]],["8.1",["8.1.0"]],["8.0",["8.0.0"]],["7.4",["7.4.2","7.4.0"]],["7.3",["7.3.0"]],["7.2",["7.2.2","7.2.1","7.2.0"]],["7.1",["7.1.1","7.1.0"]],["7.0",["7.0.3","7.0.2","7.0.1","7.0.0"]],["6.2",["6.2.2","6.2.1","6.2.0"]],["6.1",["6.1.1","6.1.0"]],["6.0",["6.0.1","6.0.0"]],["5.0",["5.0.2","5.0.1","5.0.0"]],["4.0",["4.0.0"]],["3.0",["3.0.0"]],["2.0",["2.0.0"]],["0.38",["0.38.1","0.38.0"]]]}
```

File: docs/.vuepress/plugins/dump-docs-data/canonicals.js [hunk 3]
```diff
@@ -25,6 +25,18 @@ async function generateCommonCanonicalURLs(currentCanonicals) {
     headers: fetchCommonHeaders
   });
   const docsData = await response.json();
+
+  // TODO: Mocked.
+  docsData.patches = [
```
Comment (budnix) [inline docs/.vuepress/plugins/dump-docs-data/canonicals.js:30]: This mock causes happy tests on localhost. But on production/staging, there would be some problems. Can you remove that mock?

File: docs/.vuepress/plugins/dump-docs-data/docs-versions.js [hunk 4]
```diff
@@ -22,6 +17,17 @@ async function readFromDocsLatest() {
   const response = await fetch('https://handsontable.com/docs/data/common.json');
   const data = await response.json();
 
+  // TODO: Mocked.
+  data.patches = [
```
Comment (budnix) [inline docs/.vuepress/plugins/dump-docs-data/docs-versions.js:21]: This mock causes happy tests on localhost. But on production/staging, there would be some problems. Can you remove that mock?

**Review by budnix — CHANGES_REQUESTED** [review]

File: docs/.vuepress/plugins/dump-docs-data/index.js [hunk 5]
```diff
@@ -49,10 +50,12 @@ module.exports = (options, context) => {
 
       const docsVersions = await fetchDocsVersions();
       const canonicalURLs = await generateCommonCanonicalURLs(rawCanonicalURLs);
+      const minorsLimit = docsVersions.versions[0] === 'next' ? 7 : 6;
```
Comment (budnix) [inline docs/.vuepress/plugins/dump-docs-data/index.js:53]: The versions limit list is already applied in the `docs-version.js` file. Can we stay with the idea to have defined the limitation only in that file?

File: docs/.vuepress/plugins/dump-docs-data/docs-versions.js (hunk 1 above)
Comment (wszymanski) [reply docs/.vuepress/plugins/dump-docs-data/docs-versions.js]: Changed within https://github.com/handsontable/handsontable/pull/10079/commits/e539c2a3a296ae4735d921341bed747ab79092d7 from `uniquePathes` to `uniquePatches`.

File: docs/.vuepress/plugins/dump-docs-data/docs-versions.js (hunk 2 above)
Comment (wszymanski) [reply docs/.vuepress/plugins/dump-docs-data/docs-versions.js:77]: Changed within https://github.com/handsontable/handsontable/pull/10079/commits/e539c2a3a296ae4735d921341bed747ab79092d7 by an extra `slice`.

File: docs/.vuepress/plugins/dump-docs-data/index.js (hunk 5 above)
Comment (wszymanski) [reply docs/.vuepress/plugins/dump-docs-data/index.js]: Changed within https://github.com/handsontable/handsontable/pull/10079/commits/e539c2a3a296ae4735d921341bed747ab79092d7.

File: docs/.vuepress/plugins/dump-docs-data/canonicals.js (hunk 3 above)
Comment (wszymanski) [reply docs/.vuepress/plugins/dump-docs-data/canonicals.js]: Removed within https://github.com/handsontable/handsontable/pull/10079/commits/e539c2a3a296ae4735d921341bed747ab79092d7 and https://github.com/handsontable/handsontable/pull/10079/commits/1f924816080aa8ce52bd36efb36c252675e92e20.

File: docs/.vuepress/plugins/dump-docs-data/docs-versions.js (hunk 4 above)
Comment (wszymanski) [reply docs/.vuepress/plugins/dump-docs-data/docs-versions.js]: Removed within https://github.com/handsontable/handsontable/pull/10079/commits/e539c2a3a296ae4735d921341bed747ab79092d7 and https://github.com/handsontable/handsontable/pull/10079/commits/1f924816080aa8ce52bd36efb36c252675e92e20.

**Comment by wszymanski** [comment]
> The dropdown list has a shallow indentation on mobiles. Is that is ok?

I positioned it using the CSS `rem` property and I supposed that it's okay. I can change it, but I need guidelines. WDYT @krzysztofspilka?

**Review by budnix — CHANGES_REQUESTED** [review]
@wszymanski, on the staging environment, I don't see the "next" entry in the dropdown list. I think before the PR changes, the entry was there. Can you revert that?

image

**Comment by wszymanski** [comment]
> @wszymanski, on the staging environment, I don't see the "next" entry in the dropdown list. I think before the PR changes, the entry was there. Can you revert that?
> 
> image

Thank you for your mindfulness. It has been fixed within https://github.com/handsontable/handsontable/pull/10079/commits/14c9701515e9934a97fca18e1b4e134d384a7b15.

**Review by budnix — APPROVED** [review]
👍

**Pull request merged**